hack/e2e/cleanup/cleanup_test.go

```diff
@@ -0,0 +1,248 @@
+//go:build e2e
+// +build e2e
+
+// Package cleanup-test is part of the end-to-end-tests. This package contains tests that evaluate the deletion of
+// Eventing CRs and the cascading deletion of all correlated Kubernetes resources.
+// To run the tests a k8s cluster and an Eventing-CR need to be available and configured. For this reason, the tests are
+// seperated via the 'e2e' buildtags. For more information please consult the readme.
+package cleanup_test
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"testing"
+	"time"
+
+	eventingv1alpha1 "github.com/kyma-project/eventing-manager/api/v1alpha1"
+	"github.com/kyma-project/eventing-manager/hack/e2e/env"
+	"github.com/kyma-project/eventing-manager/pkg/eventing"
+	"github.com/pkg/errors"
+	"github.com/stretchr/testify/require"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"go.uber.org/zap"
+	k8stypes "k8s.io/apimachinery/pkg/types"
+
+	"k8s.io/client-go/kubernetes"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	. "github.com/kyma-project/eventing-manager/hack/e2e/common"
+	. "github.com/kyma-project/eventing-manager/hack/e2e/common/fixtures"
+)
+
+// Constants for retries.
+const (
+	interval = 2 * time.Second
+	attempts = 60
+)
+
+// clientSet is what is used to access K8s build-in resources like Pods, Namespaces and so on.
+var clientSet *kubernetes.Clientset //nolint:gochecknoglobals // This will only be accessible in e2e tests.
+
+// k8sClient is what is used to access the Eventing CR.
+var k8sClient client.Client //nolint:gochecknoglobals // This will only be accessible in e2e tests.
+
+var logger *zap.Logger
+
+var testConfigs env.E2EConfig
+
+// TestMain runs before all the other test functions. It sets up all the resources that are shared between the different
+// test functions. It will then run the tests and finally shuts everything down.
+func TestMain(m *testing.M) {
+	var err error
+	logger, err = SetupLogger()
+	if err != nil {
+		logger.Error(err.Error())
+		panic(err)
```

Comment:
Please don't use panics, they are detected and complained by the security scanners. You can use `log.Fatal` instead.

Reply:
updated!